tests/monotouch-test/Foundation/NSDataTest.cs

```diff
@@ -155,9 +155,14 @@ public void Https ()
 				Assert.Ignore ("NSData.FromUrl doesn't seem to work in watchOS");
 			}
 #endif
+			NSError error;
 			using (var url = new NSUrl ("https://www.microsoft.com/robots.txt"))
-			using (var x = NSData.FromUrl (url)) {
-				Assert.That ((x != null) && (x.Length > 0));
+			using (var x = NSData.FromUrl (url, NSDataReadingOptions.Uncached, out error)) {
+				if (error != null) {
+					Assert.Inconclusive ($"Could not access url error is '{error.Description}'");
+				} else {
+					Assert.That ((x != null) && (x.Length > 0));
```

Comment:
Splitting the complex expression into multiple asserts would make failures easier to diagnose, since NUnit would provide better assertion messages:

```csharp
Assert.IsNotNull (x, "NotNull");
Assert.That (x, Is.Not.Empty, "NotEmpty");
```

Reply:
I like the idea of the diff urls, I'll try that, make sense, this is a network failure due to the setup of the labs, which is very annoying :/